File: src/api/session.ts
import { invoke } from '@tauri-apps/api/core';
import type { SessionTab } from '../types/session';

export const listSessionTabs = () => invoke<SessionTab[]>('list_session_tabs');

export const saveSessionTabs = (tabs: SessionTab[]) => invoke<void>('save_session_tabs', { tabs });

File: src-tauri/src/commands/session.rs
use tauri::State;

use crate::{app_state::AppState, model::SessionTab};

#[tauri::command]
pub async fn list_session_tabs(state: State<'_, AppState>) -> Result<Vec<SessionTab>, String> {
    crate::service::session_service::list_tabs(&state.pool)
        .await
        .map_err(Into::into)
}

#[tauri::command]
pub async fn save_session_tabs(
    tabs: Vec<SessionTab>,
    state: State<'_, AppState>,
) -> Result<(), String> {
    crate::service::session_service::save_tabs(&state.pool, tabs)
        .await
        .map_err(Into::into)
}
